src/Documents/Operations/Attachments/PutAttachmentOperation.ts

```diff
@@ -0,0 +1,101 @@
+import {IOperation, OperationResultType} from "../OperationAbstractions";
+import {AttachmentData, AttachmentDetails} from "../../Attachments";
+import {IDocumentStore} from "../../IDocumentStore";
+import {DocumentConventions, RavenCommand, ServerNode} from "../../..";
+import {HttpCache} from "../../../Http/HttpCache";
+import {StringUtil} from "../../../Utility/StringUtil";
+import {throwError} from "../../../Exceptions";
+import {HttpRequestParameters} from "../../../Primitives/Http";
+import * as stream from "readable-stream";
+
+export class PutAttachmentOperation implements IOperation<AttachmentDetails> {
```

Comment:
Umm what happened to the old put attachment op? I thought it was there already...

Reply:
You had BatchCommand, now we have  PutAttachmentOperation as second option to upload attachments.